webapp/channels/src/components/flag_message_modal/flag_post_modal.tsx

```diff
@@ -0,0 +1,240 @@
+    const optionalCommentSectionTitle = formatMessage({id: 'flag_message_modal.optional_comment.title', defaultMessage: 'Comment (optional)'});
+    const reasonSelectPlaceholder = formatMessage({id: 'flag_message_modal.reason_select.placeholder', defaultMessage: 'Select a reason for flagging'});
+    const commentPlaceholder = formatMessage({id: 'flag_message_modal.comment.placeholder', defaultMessage: 'Describe your concern...'});
+
+    const post = useSelector((state: GlobalState) => getPost(state, postId));
+    const channel = useSelector((state: GlobalState) => getChannel(state, post.channel_id));
+    const currentTeam = useSelector(getCurrentTeam);
+    const contentFlaggingSettings = useSelector(contentFlaggingConfig);
+
+    const reasons = useMemo(() => {
+        if (!contentFlaggingSettings || !contentFlaggingSettings.reasons) {
+            return [];
+        }
+
+        return contentFlaggingSettings.reasons.map((reason) => ({
+            value: reason,
+            label: reason,
+        }));
+    }, [contentFlaggingSettings]);
+
+    const previewMetadata: PostPreviewMetadata = useMemo(() => {
+        return {
+            post,
+            post_id: post.id,
+            team_name: currentTeam?.name || '',
+            channel_display_name: channel?.display_name || '',
+            channel_type: channel?.type || 'O',
+            channel_id: channel?.id || '',
+        };
+    }, [channel?.display_name, channel?.id, channel?.type, currentTeam?.name, post]);
+
+    // This is to bring react select dropdown above the modal body
+    const reactStyles = useMemo(() => {
+        return {
+            menuPortal: (provided) => ({
+                ...provided,
+                zIndex: 9999,
+            }),
+        } satisfies StylesConfig<SelectedOption, boolean>;
+    }, []);
+
+    const handleOptionChange = useCallback((selectedOption: SelectedOption | null) => {
+        const reason = selectedOption ? selectedOption.label : '';
+        setReason(reason);
+
+        if (reason === '') {
+            setReasonError(formatMessage({id: 'flag_message_modal.reason_required_error', defaultMessage: 'Please select a reason for flagging this message.'}));
+        } else {
+            setReasonError('');
+        }
+    }, [formatMessage]);
+
+    const handleCommentChange = useCallback((e: React.ChangeEvent<TextboxElement>) => {
+        setComment(e.target.value);
+
+        if (contentFlaggingSettings?.reporter_comment_required && e.target.value.trim() === '') {
+            setCommentError(formatMessage({id: 'flag_message_modal.empty_comment_error', defaultMessage: 'Please add a comment explaining why you’re flagging this message.'}));
+        } else {
+            setCommentError('');
+        }
+    }, [contentFlaggingSettings, formatMessage]);
+
+    const handleToggleCommentPreview = useCallback(() => {
+        setShowCommentPreview((prev) => !prev);
+    }, []);
+
+    const validateForm = useCallback((): boolean => {
```

Comment:
I guess we don't want to return early in case of an error? Are the errors appended? If not, this would causing losing the first error.

Reply:
It is intended. First it sets the `setCommentError`, then `setReasonError`, so its checking for two separate kind of errors which are displayed in different places in the modal